account_invoice_customer_no_autofollow/models/account_move.py

```diff
@@ -0,0 +1,40 @@
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def message_subscribe(self, partner_ids=None, channel_ids=None, subtype_ids=None):
+        if self.env.context.get("invoice_no_auto_follow"):
+            partner_ids.remove(self.partner_id.id)
```

Comment:
```suggestion
            if partner_ids and self.partner_id.id in partner_ids:
                partner_ids.remove(self.partner_id.id)
```

I managed to get the following exception here. A test for the case where the partner is not in the list would probably make sense. This seems to be the case when the partner was added as follower before posting the invoice.

```
Odoo Server Error
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 624, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 14, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 669, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 350, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 915, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 1343, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 1331, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 387, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 374, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons_custom/account_invoice_customer_no_autofollow/models/account_move.py", line 31, in action_post
    .get_param("account.no_auto_follow", False)
  File "/usr/lib/python3/dist-packages/odoo/addons/account/models/account_move.py", line 2362, in action_post
    return self.post()
  File "/usr/lib/python3/dist-packages/odoo/addons_custom/sale_commission/models/account_move.py", line 35, in post
    return super().post()
  File "/usr/lib/python3/dist-packages/odoo/addons_ecoservice/ecoservice_financeinterface_datev/models/account/account_move.py", line 67, in post
    return super().post()
  File "/usr/lib/python3/dist-packages/odoo/addons_ecoservice/ecoservice_financeinterface/models/account/account_move.py", line 81, in post
    return super().post()
  File "/usr/lib/python3/dist-packages/odoo/addons/account_asset/models/account_move.py", line 36, in post
    res = super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons/sale/models/account_invoice.py", line 81, in post
    res = super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons/purchase_stock/models/account_invoice.py", line 182, in post
    return super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons/l10n_de/models/datev.py", line 35, in post
    return super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons/account_reports/models/account_activity.py", line 83, in post
    return super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons/stock_account/models/account_move.py", line 52, in post
    res = super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons_ecoservice/ecoservice_partner_account-13/models/account/account_move.py", line 13, in post
    return super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons/account_invoice_extract/models/account_invoice.py", line 288, in post
    res = super(AccountMove, self).post()
  File "/usr/lib/python3/dist-packages/odoo/addons/account/models/account_move.py", line 2295, in post
    move.message_subscribe([p.id for p in [move.partner_id] if p not in move.sudo().message_partner_ids])
  File "/usr/lib/python3/dist-packages/odoo/addons_custom/account_invoice_customer_no_autofollow/models/account_move.py", line 9, in message_subscribe
    partner_ids.remove(self.partner_id.id)
ValueError: list.remove(x): x not in list
```

Reply:
Thank you for this hint! Missed it somehow. Would appreciate your review when fixed